src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -320,8 +449,9 @@ private Task InvokeStreamCore(string methodName, InvocationRequest irq, object[]
             if (!_receivedHandshakeResponse)
             {
                 // process handshake and return left over data to parse additional messages
-                if (!ProcessHandshakeResponse(ref currentData))
+                if (!ProcessHandshakeResponse(ref currentData, out var exception))
                 {
+                    await OnClosed(_connection, exception);
```

Comment:
Is this to keep closing more centralized?

Reply:
Yeah, and because `OnClosed` is async and `ProcessHandshakeResponse` has a `ref` param ;)